app/controllers/server.rb

```diff
@@ -1,9 +1,40 @@
 module RushHour
   class Server < Sinatra::Base
+    set :show_exceptions, :after_handler
+
     not_found do
       erb :error
     end
-    
-    # Write the rest of your controller code here!
+
+    post '/sources' do
+      ValidateParams.source_response(params)
+    end
+
+    post '/sources/:IDENTIFIER/data' do
+      ValidateParams.identifier_data_response(params)
```

Comment:
I love that you've extracted the accepting of data into `ValidateParams`. This is an awesome first step 👍 I'd suggest choosing a better name for this and for the methods you call from this controller.